### PR TITLE
Re-introduce support for reporting line/column/section numbers in MS Word via UIA, using UIA remote ops

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
This reintroduces the commits in pr #13283 which had been reverted in pr #13350. However it also addresses build issues.

### Summary of the issue:
PR #13283  introduced support for reporting line/column/section numbers in Microsoft word via UI Automation, by using the UI Automation Remote Operations Library.
However, this PR had to be reverted in #13350  as
1. UIA would not initialize properly in binary builds due to a missing manifest file. (#13347)
2. NVDA failed to build on Visual Studio 2022. Specifically when building the Remote Ops library with msbuild, midl would fail with an error about a system environment variable being missing.
3. NVDA could no longer be built on Windows 7. (#13346)

### Description of how this pull request fixes the issue:
This PR reintroduces all of the original changes, but:
* setup.py now includes *.manifest files in the lib directory.
* NVDA again now requires Visual Studio 2019. However, builds will not fail if a newer version of Visual Studio (E.g. 2022) is installed along side 2019. this is managed by setting MSVC_VERSION in scons before it looks for Visual Studio, so that it specifically selects VS 2019 (VC 14.2).
Although building on Windows 7 could not be fixed, the readme now notes that only building on Windows 10 and higher is supported.

### Testing strategy:
* Ran tests in pr #13283 
* Successfully built NVDA from source on a machine with both VS 2019 and VS 2022 installed

### Known issues with pull request:
Pr #13033 allowed NVDA to be built with Visual Studio 2022, though keeping VS 2019 as the official version (which is used on Appveyor). This PR again limits building to Visual Studio 2019. Obviously we do wish to support VS 2022 in the future, but as the support was recently introduced, it was not official, and the UIA Remote Ops library does not yet support VS 2022, we had to make the decision to drop this support for now. At least until we have time to find a work around in NVDA or get the underlying issue fixed in the Remote Ops library.
However, unlike before, it is at least now possible to build NVDA with both VS 2019 and 2022 installed at the same time which should solve some of the original issues.
Although NVDA can run on Windows 7 Service Pack 1 and higher, NVDA cannot be built on Windows 7. Windows 10 or higher is now required.

### Change log entries:
Already included in pr.
New features
Changes
Bug fixes
For Developers

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Review by lukaszgo1 — CHANGES_REQUESTED** [review]
Given that merging this would reintroduce #13346 it would be nice to document in the readme that Windows 7 is no longer a supported platform for building NVDA. Please also document the newly added sub-modules in the readme. While at it it would be good to explain why you're using your own fork of UIA library.

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -69,7 +69,7 @@ What's New in NVDA
 
 == Changes for Developers ==
 - Note: this is a Add-on API compatibility breaking release. Add-ons will need to be re-tested and have their manifest updated.
-- Compiling NVDA dependencies with Visual Studio 2022 (17.0) is now supported. For development and release builds, Visual Studio 2019 is still used. (#13033)
+- Although NVDA still requires Visual Studio 2019, Builds should no longer fail if a newer version of Visual Studio (E.g. 2022) is installed along side 2019.
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:72]:
```suggestion
- Although NVDA still requires Visual Studio 2019, Builds should no longer fail if a newer version of Visual Studio (E.g. 2022) is installed along side 2019. (#13033, #13387)
```

**Review by seanbudd — COMMENTED** [review]
I have reviewed this PR from 663227546b7c54d2bea2105d8ada567261659c27 [filtered view](https://github.com/nvaccess/nvda/pull/13387/files/e9ae64fdfa4ee7186c235aeec575ca941ca57cca..e4290ea9e86f91e323ce945e821f47b3e4e8e6ae).

LGTM, other than fixing the changelog entry. I am assuming the earlier commits from the reverted PR do not need to be re-reviewed.

File: user_docs/en/changes.t2t [hunk 2]
```diff
@@ -68,7 +69,7 @@ What's New in NVDA
 
 == Changes for Developers ==
 - Note: this is a Add-on API compatibility breaking release. Add-ons will need to be re-tested and have their manifest updated.
```
Comment (XLTechie) [inline user_docs/en/changes.t2t:71]: "a add-on" s/b "an add-on", and "manifest" s/b "manifests". Also, should the first "Add-on" be capitalized?

```suggestion
- Note: this is an add-on API compatibility breaking release. Add-ons will need to be re-tested and have their manifests updated.

```

**Review by seanbudd — APPROVED** [review]
LGTM

**Comment by leonardder** [comment]
Ugh, slightly unhappy about the very fast way this pull request went through, as I didn't have the time to suggest a sane alternative for the Visual Studio problem. In fact, only installing the MSVC 142 toolset in VS2022 should have been enough. That also clarifies why NVDA built successfully on appveyor when it was still using VS2022, since the 142 (2019) toolset is installed in the image. Now, building is only supported when VS 2019 is installed alongside 2022, which is a no go for me.

**Comment by leonardder** [comment]
AH, never mind my initial comment, it was based on wrong assumptions. I'm sorry, probably wasn't entirely awake yet.

**Comment by michaelDCurran** [comment]
@leonardder and you're on Windows 11 with use UIA to access MS Word NVDA setting enabled?
This may be an issue with this feature not being necessarily rolled out to all installs of MS Word perhaps. 
I could look at providing you with a try build with greater logging to help debug if you like.
Certainly working on my machine though.

**Comment by michaelDCurran** [comment]
@leonardder  Do you see the following lines in your log:
```
INFO - NVDAHelperLocal (19:43:26.419) - UIAHandler.UIAHandler.MTAThread (15144):
Thread 15144, build\x86\UIARemote\UIARemote.cpp, initialize, 145:
Registered C:\Program Files (x86)\NVDA\lib\alpha-24809,2583f8fe\Microsoft.UI.UIAutomation.dll.manifest

INFO - NVDAHelperLocal (19:43:26.436) - UIAHandler.UIAHandler.MTAThread (15144):
Thread 15144, build\x86\UIARemote\UIARemote.cpp, initialize, 150:
Microsoft.UI.UIAutomation is available
```

**Comment by leonardder** [comment]
Yes, these log lines are in my log and UIA in word is enabled. Some logging might be helpful indeed. Alternatively, may be I should switch to another release channel?

**Comment by leonardder** [comment]
> This may be an issue with this feature not being necessarily rolled out to all installs of MS Word perhaps.

I think this is the cause indeed. I was on the current branch and now updated to the current (preview) branch, build 16.0.14931.20094. All works fine now.

**Pull request merged**